admin/controllers/school.js

```diff
@@ -444,14 +444,47 @@ const getSelectPupilNotTakingCheck = async (req, res, next) => {
 
   return res.render('school/select-pupils-not-taking-check', {
     breadcrumbs: req.breadcrumbs(),
+    sortField,
+    sortDirection,
     attendanceCodes,
-    formData,
     pupilsList,
     htmlSortDirection,
     arrowSortDirection
   })
 }
 
+const savePupilNotTakingCheck = async (req, res, next) => {
+  res.locals.pageTitle = 'Save pupils not taking the check'
+  req.breadcrumbs(res.locals.pageTitle)
+
+  if (req.body.attendanceCode === undefined || req.body.pupil === undefined) {
+    return res.redirect('/school/pupils-not-taking-check/select-pupils')
+  }
+
+  const todayDate = moment(moment.now()).format()
+  let pupils = req.body.pupil
+
+  pupils.forEach(async (id) => {
+    let pupil = await fetchOnePupil(id, req.user.School)
+    if (pupil) {
+      pupil.attendanceCode = {
+        _id: req.body.attendanceCode,
+        dateRecorded: new Date(todayDate),
+        byUser: req.user.UserName
+      }
+      await pupil.save()
```

Comment:
Equally adding a save inside the loop is not recommended.  Please change the flow of execution to a) fetch all pupils, b) process the pupils (the loop), and then c) save the pupils.

Reply:
@jon-shipley Modified as suggested to insertMany outside of the loop.